client/src/components/StreamField/StreamField.scss

```diff
@@ -1,19 +1,28 @@
 $grid-gutter-width: 30px;
-$header-gutter: 8px;
-$block-margin-vertical: theme('spacing.1');
-$block-margin-horizontal: theme('spacing.3');
 $type-button-padding-vertical: 10px;
 $type-button-padding-horizontal: 10px;
-$content-padding-horizontal: 24px;
-$content-padding-vertical: 16px;
-$action-font-size: 1.125rem;
 $add-transition-duration: 0.3s;
-$hover-transition-duration: 0.3s;
 $border-radius: 3px;
 $add-panel-gutter: 8px;
 
 @import 'scss/components/c-sf-add-button';
 @import 'scss/components/c-sf-add-panel';
 @import 'scss/components/c-sf-button';
-@import 'scss/components/c-sf-block';
-@import 'scss/components/c-sf-container';
+
+[aria-expanded='true'] + .w-panel__heading .c-sf-block__title {
+  @apply w-sr-only;
+}
+
+// Only hide the block type if the field is collapsed *and* the title isn’t empty.
+[aria-expanded='false']
+  + .w-panel__heading
+  .c-sf-block__title:not(:empty)
+  + .c-sf-block__type {
+  display: none;
+}
```

Comment:
This selector is a bit frightening, but doing it like this in CSS saves quite a bit of JS.

Reply:
Good tradeoff IMO